pkg/cmd/roachprod-microbench/cluster/execute.go

```diff
@@ -77,13 +140,32 @@ func remoteWorker(
 				}
 				break
 			}
+
+			// Check if the command's group has already been cancelled
+			if command.GroupID != "" {
```

Comment:
This check technically isn't needed right? If the group is cancelled the `withGroupCancellation` context will get immediately cancelled as well which is caught by:

```
	if errors.Is(cmdCtx.Err(), context.Canceled) {
				responseChan <- RemoteResponse{RemoteCommand: command, commandStatus: Cancelled}
				continue
			}
```

below. Although maybe it could race, so no harm in keeping it 🤷